migrate.py

```diff
@@ -1933,6 +1935,31 @@ def _rewrite_yaml_mapping(el, namespace, collection, spec, args, dest, checkout_
 
 def _rewrite_yaml_mapping_keys_non_vars(el, namespace, collection, spec, args, dest):
     translate = []
+
+    m = ModuleArgsParser(el)
+    try:
+        module_name, module_args, to = m.parse()
+
+        for ns in spec.keys():
```

Comment:
Could also optimize the loop like this:
```suggestion
        for ns in (() if module_name else spec.keys()):
```